### PR TITLE
pipcook2.0: dataSource add scriptQuery

### PR DESCRIPTION
**Pull request by Ahkari**
datasource querystring supported

### CODE REVIEW

**Comment by FeelyChau** [comment]
Thank you for your PR,  and please run `yarn lint` before commit. @Ahkari

File: packages/core/src/types/pipeline.ts [hunk 1]
```diff
@@ -27,6 +27,13 @@ export interface PipelineMeta {
  */
 export enum ScriptType { DataSource, Dataflow, Model }
 
+/**
+ * script query stucture
+ */
+export interface ScriptQuery<T = string> {
+  [key: string]: T | T[] | null;
+}
```
Comment (WenheLI) [inline packages/core/src/types/pipeline.ts:35]: `query` should be string only.  Generic may not be needed here

**Review by WenheLI — CHANGES_REQUESTED** [review]

File: packages/cli/src/utils/script.ts [hunk 2]
```diff
@@ -8,17 +8,20 @@ import {
 import * as path from 'path';
 import { URL } from 'url';
 import { fetchWithCache } from './cache';
+import * as queryString from 'query-string';
 
 export const downloadScript = async (scriptDir: string, scriptOrder: number, url: string, type: ScriptType, enableCache = true): Promise<PipcookScript> => {
   const urlObj = new URL(url);
   const baseName = path.parse(urlObj.pathname).base;
   const localPath = path.join(scriptDir, `${scriptOrder}-${baseName}`);
+  const { query } = queryString.parseUrl(url);
```
Comment (FeelyChau) [inline packages/cli/src/utils/script.ts:17]:
```suggestion
  const { query } = queryString.parse(urlObj.query);
```

**Review by FeelyChau — CHANGES_REQUESTED** [review]

**Comment by FeelyChau** [comment]
LGTM.

**Review by FeelyChau — APPROVED** [review]

**Comment by yorkie** [comment]
@Ahkari May I ask you to correct the title?

**Review by WenheLI — APPROVED** [review]

**Pull request merged**